packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -319,4 +312,52 @@ impl<C: AsClient> EntityStore for PostgresStore<C> {
 
         Ok(metadata)
     }
+
+    async fn archive_entity(
+        &mut self,
+        entity_id: EntityId,
+        actor_id: UpdatedById,
+    ) -> Result<(), ArchivalError> {
+        let mut transaction = PostgresStore::new(
+            self.as_mut_client()
+                .transaction()
+                .await
+                .into_report()
+                .change_context(ArchivalError)?,
+        );
+
+        transaction
+            .lock_entity_for_update(entity_id)
+            .await
+            .change_context(ArchivalError)?;
+
+        let previous_entity: PersistedEntity = transaction
+            .read_one(&Filter::for_latest_entity_by_entity_id(entity_id))
+            .await
+            .change_context(ArchivalError)?;
+
+        transaction
+            .update_entity(
+                entity_id,
+                previous_entity.inner().clone(),
+                previous_entity.metadata().entity_type_id().clone(),
+                actor_id,
+            )
+            .await
+            .change_context(ArchivalError)?;
+
+        transaction
+            .move_entity_to_histories(entity_id, true)
```

Comment:
Should we make this bool an Enum?

Reply:
I'd say yes, this is more clear. My only concern is that naming boolean enums is hard 😞

Reply:
How do you feel about this, 816781b2772e62aeb5a3a276395499c6bf855a25 @TimDiekmann ?

Reply:
Unsure about the `As*`, but definitely better. I'm fine with keeping the `As*`, though.

However, lets derive common traits, `#[derive(Debug, Copy, Clone, PartialEq, Eq)]` and use `==` instead of `matches!` 🙂

Reply:
Derives in 3ec511d74be1cf33696974d4a2aa83e4960c55de

Reply:
Thanks!